.github/workflows/integration.yml

```diff
@@ -226,26 +228,39 @@ jobs:
           script: |
             const log = github.log
             return {
+              runnerDebugMode: core.isDebug(),
               debug: log.debug === console.debug,
               info: log.info === console.info
             }
-      - run: |
-          echo "- Validating debug default"
-          expected='{debug:false,info:false}'
-          if [[ "${{steps.debug-default.outputs.result}}" != "$expected" ]]; then
-            echo $'::error::\u274C' "Expected '$expected', got ${{steps.debug-default.outputs.result}}"
-            exit 1
-          fi
-          echo "- Validating debug set to true"
-          expected='{debug:true,info:true}'
-          if [[ "${{steps.debug-true.outputs.result}}" != "$expected" ]]; then
-            echo $'::error::\u274C' "Expected '$expected', got ${{steps.debug-true.outputs.result}}"
-            exit 1
-          fi
-          echo "- Validating debug set to false"
-          expected='{debug:false,info:false}'
-          if [[ "${{steps.debug-false.outputs.result}}" != "$expected" ]]; then
-            echo $'::error::\u274C' "Expected '$expected', got ${{steps.debug-false.outputs.result}}"
-            exit 1
-          fi
+      - id: evaluate-tests
+        name: Evaluate test outputs
+        env:
+          RDMODE: ${{ (runner.debug == '1' && 'true') || 'false' }}
```

Comment:
May sound wild, but is it possible to utilize https://docs.github.com/en/rest/actions/secrets?apiVersion=2022-11-28#create-or-update-a-repository-secret to temporarily set the secret?

Reply:
Another wild idea: create a reusable workflow with a secret name identical to the built-in one, and call it with different values.
See https://docs.github.com/en/actions/using-workflows/reusing-workflows#passing-inputs-and-secrets-to-a-reusable-workflow

Reply:
That'd be extremely race-sensitive; one workflow run sets the secret while another is also running and so we screw up the tests for that one.

However, I'm pretty sure you could [trigger a re-run of a workflow job](https://docs.github.com/en/rest/actions/workflow-runs?apiVersion=2022-11-28#re-run-a-workflow), which allows you to enable debugging. It probably has to be done from a separate workflow, one that triggers when the integration workflow completed, perhaps.

Reply:
Although, I must say I'm not certain that the re-run will be presented very coherently. Instead, I'd experiment with putting the `ACTIONS_STEP_DEBUG` secret in an _environment_ and have a workflow reference that environment via `jobs.<job-id>.environment`. The environment would need to be created and left 'unprotected', so no protection rules, but that woud give you a workflow-specific secret.

Now, if the environment could be selected in a _matrix_ then that'd be even better!

Side note: The documentation half suggest that `ACTIONS_STEP_DEBUG` doesn't have to be a secret, it can be a repository *variable* too (so, set in the same place as action secrets, just not encrypted).

Reply:
Variables are a new thing, be careful: https://docs.github.com/en/actions/learn-github-actions/contexts#vars-context

I think the env is the best course here, and I would assume to be able to select from a matrix, because it just makes sense to run the same job in different envs (e.g. integration tests or deployments).

Reply:
Ah, didn't know that vars are still subject to change.

I think there is a bit of confusion here: `job.<job-id>.environment` is **not the same thing** as `env` environment variables. `environment` names a _deployment_ environment, the only reason to use this here is to (ab)use the fact that can select a specific deployment environment in a workflow and so limit secrets to that specific workflow. Otherwise, the `ACTIONS_STEP_DEBUG` secret would apply to **all** workflows in this repository.

At issue here is that we can't use an `env` environment variable to switch the value of `runner.debug`, and that's the only way that you can detect if runner debug mode is enabled in `actions.yml` a `inputs.<input-name>.default` expression, the **only** option that is left is setting the `ACTIONS_STEP_DEBUG` secret, but you'd want to do so only for a workflow running the integration test.

Reply:
> as env environment variables. environment names a deployment environment

All clear, we're on the same page. Sorry, I was just lazy to write the whole word :) I realize now that they both mean specific things in this world 🤦‍♂️.

Reply:
> Another wild idea: create a reusable workflow with a secret name identical to the built-in one, and call it with different values.

That was a great idea, except for this:

> : secret name `ACTIONS_STEP_DEBUG` within `workflow_call` can not be used since it would collide with system reserved name

See https://github.com/actions/github-script/actions/runs/4620133014

Reply:
In the meantime tried the environment one: https://github.com/TWiStErRob/github-actions-test/pull/10

Reply:
So, environments work, but environments can't be set dynamically, and you can't use `environment` in a re-usable workflow call either. So this would require a separate workflow job that references the environment and runs the `debug` test again, now with `runner.debug` enabled.

If you give me an environment name, I can add the extra job definition?

Reply:
And in the meantime, I've also [posted a community question on the subject](https://github.com/orgs/community/discussions/52079).

Reply:
Interesting learning is that you can set the debug flags with vars/secrets, and they'll be **only** available in those contexts, so the only surefire way to detect step debugging is `runner.debug` which is only mirroring (`ACTIONS_STEP_DEBUG` btw, not `ACTIONS_RUNNER_DEBUG` 😱; oh naming...)

Also precedence seems to be (first wins):
 - `secrets.ACTIONS_STEP_DEBUG`
 - [ ] Enable debug logging  
   Sets `secrets.ACTIONS_STEP_DEBUG`, but only if not already set to some value.
 - `vars.ACTIONS_STEP_DEBUG`

Reply:
> If you give me an environment name, I can add the extra job definition?

I'm not a maintainer here. But from my experiment if you run in a non-existent environment, it'll be just created as empty :)

Reply:
> You can't dynamically set the environment name used for a workflow job,

Isn't that what I did here? https://github.com/TWiStErRob/github-actions-test/pull/10/files#diff-f3ad4bc1e502b99a000fd651783c4e17cc5f8a0ab5e50184fc880221ce0a1f27R11

Reply:
👋 I created a `debug-integration-test` environment with `ACTIONS_STEP_DEBUG` as a variable set to `true`

Reply:
Uhm, unless that part has to evaluate the workflow environment context *after* it has been processsed by something outside of the action runner, and so must avoid re-evaluating expression output. :facepalm:

Reply:
It picks up the right vars/secrets regardless of being a literal or not :)

https://github.com/TWiStErRob/github-actions-test/actions/runs/4620544892

Reply:
It clearly does pick up the environment:

<img width="731" alt="image" src="https://user-images.githubusercontent.com/46775/230142539-75905683-6e82-477d-ad7d-ef9a16f223a9.png">

Reply:
> 👋 I created a debug-integration-test environment with ACTIONS_STEP_DEBUG as a **variable** set to true

@joshmgross did you set `vars` or `secrets`

Reply:
I set `vars` 🤔

I'd guess your PR doesn't have access to variables as a fork, let me double check our policy there

Reply:
I set a secret in my own fork repo, and https://github.com/mjpieters/github-script/actions/runs/4620603280/jobs/8171064413 shows that that at least works.

Reply:
Yeah we don't support this yet, but possibly soon we'll be able to send variables to fork PRs.